snowflake/src/integration/java/org/apache/iceberg/snowflake/NamespaceTests.java

```diff
@@ -0,0 +1,290 @@
+                      Namespace.of(dbName, "INFORMATION_SCHEMA"))));
+    } finally {
+      dropSchemaIfExists(schema1);
+      dropSchemaIfExists(schema2);
+    }
+  }
+
+  @Test
+  public void testListNamespacesAtSchemaLevelIsNotAllowed()
+      throws SQLException, InterruptedException {
+    String schema1 = "Schema_1";
+    String dbName = TestConfigurations.getInstance().getDatabase().toUpperCase();
+    try {
+      clientPool.run(conn -> conn.createStatement().execute("use " + dbName));
+      createOrReplaceSchema(schema1);
+      Assertions.assertThatException()
+          .isThrownBy(
+              () -> {
+                snowflakeCatalog.listNamespaces(Namespace.of(dbName, schema1));
+              })
+          .isInstanceOf(IllegalArgumentException.class);
+    } finally {
+      dropSchemaIfExists(schema1);
+    }
+  }
+
+  @Test
+  public void testLoadNonExistingRootLevelNamespace() {
+    String nonExistingDb = "IDontExist";
+    Assertions.assertThatThrownBy(
+            () -> {
+              snowflakeCatalog.loadNamespaceMetadata(Namespace.of(nonExistingDb));
+            })
+        .isInstanceOf(NoSuchNamespaceException.class);
+  }
+
+  @Test
+  public void testLoadNonExistingDBLevelNamespace() {
+    String nonExistingSchema = "IDontExist";
+    String db = TestConfigurations.getInstance().getDatabase();
+    Assertions.assertThatThrownBy(
+            () -> {
+              snowflakeCatalog.loadNamespaceMetadata(Namespace.of(db, nonExistingSchema));
+            })
+        .isInstanceOf(NoSuchNamespaceException.class);
+  }
+
+  @Test
+  public void testLoadNamespaceThatExceedMaxSupportedHierarchy() {
+    String nonExistingSchema = "IDontExist";
+    String db = TestConfigurations.getInstance().getDatabase();
+    Assertions.assertThatThrownBy(
+            () -> {
+              snowflakeCatalog.loadNamespaceMetadata(
+                  Namespace.of(db, nonExistingSchema, "UnSupportedLevel"));
+            })
+        .isInstanceOf(IllegalArgumentException.class);
+  }
+
+  @Test
+  public void testLoadNamespaceWithUnquotedIdentifier() throws SQLException, InterruptedException {
+    String schema1 = "Schema123";
+    String dbName = TestConfigurations.getInstance().getDatabase();
+    try {
+      Assertions.assertThatNoException()
+          .isThrownBy(
+              () -> {
+                clientPool.run(conn -> conn.createStatement().execute("use " + dbName));
+                createOrReplaceSchema(schema1);
+                snowflakeCatalog.loadNamespaceMetadata(Namespace.of(dbName, schema1));
+              });
+    } finally {
+      dropSchemaIfExists(schema1);
+    }
+  }
+
+  @Test
+  public void testLoadNamespaceWithUnquotedIdentifierIsCaseInsensitive()
+      throws SQLException, InterruptedException {
+    String schema1 = "Schema123";
+    String dbName = TestConfigurations.getInstance().getDatabase();
+    try {
+      Assertions.assertThatNoException()
```

Comment:
same as above, it's better to verify that the correct namespace metadata is returned